src/components/Header/docs.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import {
+  ArgsTable,
+  Description,
+  Heading,
+  Source,
+  Stories,
+  Subheading,
+} from '@storybook/addon-docs/blocks';
+
+import { DocsTable } from '../internal/DocsTable';
+
+import { Header } from './';
+
+export const docs = {
+  page: () => (
+    <>
+      <Description>
+        The header component is used for navigation on a site. It is designed to be versatile and
+        responsive, rendering a full menu on large screens and collapsing into a drawer on small
+        screens.
+      </Description>
+      <Heading>Props</Heading>
+      <ArgsTable of={Header} />
+
+      <Subheading>HeaderItem</Subheading>
+      <Description>An item in the header.</Description>
+      <DocsTable
+        data={[
+          {
+            name: (
+              <>
+                children
+                <DocsTable.Asterisk />
+              </>
+            ),
+            type: <code>HeaderChildItem[]</code>,
+            default: '',
+            description: 'The content of the header item.',
+          },
+          {
+            name: 'disabled',
+            type: <code>boolean</code>,
+            default: <code>false</code>,
+            description: (
+              <>
+                If <code>true</code>, the element will be unusable.
+              </>
+            ),
+          },
+          {
+            name: (
+              <>
+                element
+                <DocsTable.Asterisk />
+              </>
+            ),
+            type: <code>node</code>,
+            default: '',
+            description: 'The content of the header item.',
+          },
+          {
+            name: 'htmlTag',
+            type: <code>HTMLAttribute</code>,
+            default: '',
+            description: 'The HTML tag to cast this element into.',
+          },
+          {
+            name: 'styled',
+            type: <code>boolean</code>,
+            default: <code>false</code>,
+            description: (
+              <>
+                If <code>true</code>, the element will be rendered as-is with no styling applied by
+                the header component.
+              </>
+            ),
+          },
+        ]}
+      />
+      <Description>
+        Header items also accept most HTML attributes as props. This means we can make a header item
+        any HTML tag, with any HTML attributes, like this:
+      </Description>
+      <Source
+        code={`
+<Header
```

Comment:
By instinct I fell that we need some indents. Will indents influence the final outputs here?

Reply:
I found that indenting the code actually makes the output indented incorrectly. This renders it correctly, but yes, the code looks a bit ugly. I think there are libraries we can use to solve this, if you think it would be better, e.g. https://www.npmjs.com/package/ts-dedent

<img width="232" alt="image" src="https://user-images.githubusercontent.com/15721063/103751460-37a2b700-506d-11eb-8370-c7182e6833f4.png">

Reply:
Thanks for it 👍. I think it is fine leave it for now. It is still pretty readable.